### DISCUSSION TITLE
RTC DS1307

### DISCUSSION THREAD

**Discussion by rj10rd** [arendst/Tasmota#14381] [Q&A] [answered]
Tasmota support ds1307 Rtc clock?

**Comment by barbudor** [comment]
https://tasmota.github.io/docs/I2CDEVICES/#supported-i2c-devices

No.
Only RTC supported is DS3221

**Reply by sfromis** [reply]
Or you can use a supported GPS as a RTC module....

**Comment by rj10rd** [comment, marked as answer]
Please see this

https://github.com/arendst/Tasmota/pull/4029#commitcomment-30886890

**Reply by ascillato2** [reply]
Please, can you test that modification?

**Reply by rj10rd** [reply]
Ok I will tell.

**Reply by rj10rd** [reply]
> Please, can you test that modification?

I tasted. It is working properly..
Tasmota is awesome

**Reply by stefan-oskamp** [reply]
The DS3231 code mentions that DS1307 is also supported. However the chips differ slightly and the DS3231 code uses registers that don't exist in the DS1307 and it uses common registers in a way that is not compatible with their definition in the DS1307 data sheet.

I have added support for the DS1307, but I need to get the right battery before I can fully test the RTC functionality and subsequently submit a pull request, if that is welcome.

Did anybody test the DS3231 code with a DS1307 and a battery?

**Reply by arendst** [reply]
Yep. My DS1307 breakout from Sparkfun works great using the current code. Don't add features you won't use but adds code size. (https://www.antratek.nl/real-time-clock-module)

**Reply by stefan-oskamp** [reply]
That is interesting because the registers are really different. See here for the DS1307:
![image](https://user-images.githubusercontent.com/57906326/209694222-54079853-5f2f-4151-9b3a-8d4ef1600f5e.png)
The detection code reads the DS3231's status register (0x0F) which is just RAM in the DS1307. Also, the code writes to the status register to clear the oscillator stop flag. This does nothing in the DS1307 except storing the status in a RAM register.
Further, the DS1307 does not store the century. The current code should return 22 for the DS1307 as the year in the TIME_T struct instead of 122 (years since 1900).
I understand your focus on code size, but I just wanted to let you know. If interested I can paste the DS1307-specific code.

**Reply by arendst** [reply]
The current code perfectly detects the DS1307 without knowing if it's a DS1307 or DS3231. The functionality is there too so why making all the hussle of implementing centuries and oscillators stop/start if it is not needed.

That's what I meant with adding things to the code that may be by the book but are of no relevance to the functionality.

Having only the year within a century is no problem at all. The correct century is added and once the century turns over to 2100 Tasmota is long dead and burried.
